src/main/java/ch/njol/skript/expressions/arithmetic/ExprArithmetic.java

```diff
@@ -328,6 +333,28 @@ public Class<? extends T> getReturnType() {
 		return returnType;
 	}
 
+	@Override
+	public Class<? extends T>[] possibleReturnTypes() {
+		if (returnType == Object.class)
+			//noinspection unchecked
+			return knownReturnTypes.toArray(new Class[0]);
+		return super.possibleReturnTypes();
+	}
+
+	@Override
+	public boolean canReturn(Class<?> returnType) {
+		if (this.returnType == Object.class) {
+			if (knownReturnTypes.contains(returnType))
+				return true;
+			for (Class<?> type : knownReturnTypes) {
+				if (returnType.isAssignableFrom(type))
+					return true;
+			}
+			return false;
```

Comment:
Can't we just call `super.canReturn()` for this?

Reply:
I think we can just cut out a chunk here, actually.